insights/tests/client/connection/test_test_connection.py

```diff
@@ -0,0 +1,175 @@
+from logging import ERROR
+
+from mock.mock import patch
+from pytest import fixture
+from pytest import mark
+from pytest import raises
+from requests.exceptions import ConnectTimeout
+from requests.exceptions import ReadTimeout
+from requests.exceptions import SSLError
+
+from insights.client.config import InsightsConfig
+from insights.client.connection import InsightsConnection
+
+LOGGING_FILE = "logging-file.log"
+
+EXPECTED_EXCEPTIONS = [SSLError, ConnectTimeout, ReadTimeout]
+parametrize_exceptions = mark.parametrize(
+    ["exception"],
+    [(exception,) for exception in EXPECTED_EXCEPTIONS],
+)
+
+
+@fixture
+@patch("insights.client.connection.InsightsConnection._init_session")
+@patch("insights.client.connection.InsightsConnection.get_proxies")
+def insights_connection(get_proxies, init_session):
+    config = InsightsConfig(base_url="www.example.com", logging_file=LOGGING_FILE)
+
+    connection = InsightsConnection(config)
+    connection.proxies = None
+
+    return connection
+
+
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_test_connection_test_urls_no_catch(test_urls, insights_connection):
+    """The connection test is stopped in case of an unknown exception."""
+    test_urls.side_effect = Exception
+
+    try:
+        insights_connection.test_connection()
+    except Exception:
+        pass
```

Comment:
Would it make sense to define a custom `Exception` subclass only for this file? E.g. `class SideEffectException(Exception): pass`.

This way we could be sure that we're catching this mocked error and not something else from within the code that got raised by accident. This applies to all occurrences.

Reply:
That’s very good observation! I was thinking about that and thought it would be too much of a hassle. Will do!